docs/source/tutorials/artifact.rst

```diff
@@ -0,0 +1,201 @@
+go through how to view, delete, and write data to an artifact using the tools provided by
+:class:`~vivarium_public_health.dataset_manager.artifact.Artifact`. You'll access data in the artifact through keys,
+mirroring the underlying hdf storage of artifacts.
+
+.. contents::
+   :depth: 1
+   :local:
+   :backlinks: none
+
+Creating an artifact
+---------------------
+To view an existing hdf file via the :class:`~vivarium_public_health.dataset_manager.artifact.Artifact` tools, we'll
+create a new artifact. We can print the resulting artifact to view the tree structure of the keys
+in our artifact. We'll use our test artifact to illustrate:
+
+.. code-block:: python
+
+    from vivarium_public_health.dataset_manager import Artifact
+
+    art = Artifact('test_artifact.hdf')
+    print(art)
+
+::
+
+    Artifact containing the following keys:
+    metadata
+            keyspace
+            locations
+            versions
+    population
+            age_bins
+            structure
+            theoretical_minimum_risk_life_expectancy
+
+Now we have an :class:`~vivarium_public_health.dataset_manager.artifact.Artifact` object, which we can use to interact
+with the data stored in the hdf file with which we created it.
+
+Optionally, we can specify filter terms on the artifact when we create it, which will be applied to filter all data
+that we load out of the artifact. For example, if we were only interested in data for years after 2005, we could add a
+filter term to our artifact that would filter all data with a ``year_start`` column. Any data in our artifact that does
+not contain this column will not be filtered. If we also wanted to filter to ages under 5, we could add another filter
+term to the list as below:
+
+.. code-block:: python
+
+    from vivarium_public_health.dataset_manager import Artifact
+
+    art = Artifact('test_artifact.hdf', filter_terms=['year_start > 2005', 'age_group_start <= 5'])
+    print(art)
+
+::
+
+    Artifact containing the following keys:
+    metadata
+            keyspace
+            locations
+            versions
+    population
+            age_bins
+            structure
+            theoretical_minimum_risk_life_expectancy
+
+Note that the keys in the artifact are unchanged. The filter terms only affect data when it is loaded out of the artifact.
+
+
+Artifacts store data under keys. Each key is of the form ``<type>.<name>.<measure>``, e.g.,
+"cause.all_causes.restrictions" or ``<type>.<measure>``, e.g., "population.structure." To view all keys in an
+artifact, use the ``keys`` attribute of the
+artifact:
+
+.. code-block:: python
+
+    art.keys
+
+::
+
+    [EntityKey(metadata.keyspace), EntityKey(metadata.locations), EntityKey(metadata.versions), EntityKey(population.age_bins),
+     EntityKey(population.structure), EntityKey(population.theoretical_minimum_risk_life_expectancy)]
+
+What we get back is a list of :class:`~vivarium_public_health.dataset_manager.artifact.EntityKey` objects. We can
```

Comment:
Note that this is an internal representation of the keys, and that users can just use strings.